newIDE/app/src/GameDashboard/ShareDialog.js

```diff
@@ -0,0 +1,148 @@
+// @flow
+import * as React from 'react';
+import { Trans } from '@lingui/macro';
+
+import CircularProgress from '@material-ui/core/CircularProgress';
+import Share from '@material-ui/icons/Share';
+
+import Dialog from '../UI/Dialog';
+import FlatButton from '../UI/FlatButton';
+import { Column, Line } from '../UI/Grid';
+import InfoBar from '../UI/Messages/InfoBar';
+import ShareButtons from '../UI/ShareDialog/ShareButtons';
+import ShareLink from '../UI/ShareDialog/ShareLink';
+
+import AuthenticatedUserContext from '../Profile/AuthenticatedUserContext';
+import {
+  type GameSlug,
+  type Game,
+  getGameSlugs,
+  getGameUrl,
+} from '../Utils/GDevelopServices/Game';
+import Window from '../Utils/Window';
+import AlertMessage from '../UI/AlertMessage';
+
+type Props = {| game: Game, onClose: () => void |};
+
+const ShareDialog = ({ game, onClose }: Props) => {
+  const [isFetchingGameSlug, setIsFetchingGameSlug] = React.useState(true);
+  const [gameSlug, setGameSlug] = React.useState<?GameSlug>(null);
+  const [showCopiedInfoBar, setShowCopiedInfoBar] = React.useState(false);
+  const [showAlertMessage, setShowAlertMessage] = React.useState(false);
+  const { getAuthorizationHeader, profile } = React.useContext(
+    AuthenticatedUserContext
+  );
+  const fetchGameSlug = React.useCallback(
+    async () => {
+      if (!profile) return;
+      try {
+        const gameSlugs = await getGameSlugs(
+          getAuthorizationHeader,
+          profile.id,
+          game.id
+        );
+        setGameSlug(gameSlugs[0]);
+      } catch (error) {
+        console.log(error);
+        setShowAlertMessage(true);
+      } finally {
+        setIsFetchingGameSlug(false);
+      }
+    },
+    [getAuthorizationHeader, profile, game]
+  );
+
+  React.useEffect(
+    () => {
+      fetchGameSlug();
+    },
+    [fetchGameSlug]
+  );
+
+  const gameUrl = getGameUrl(game, gameSlug);
+  if (!gameUrl) return null;
+
+  const onCopyLinkToClipboard = () => {
```

Comment:
you seem to have this logic copied twice (here and in OnlineGameLink + onOpen + onShare.
Should they be integrated inside the <ShareLink> component?
The `ShareLink` could just take a `url` and create the textfield with the buttons & the actions associated

Reply:
There is a bit of logic involved in `OnlineGameLink` on the display of ShareButtons (it relies on whether the game is published or not), so I could only factorise a `ShareButton` (and rename `ShareButtons` as `SocialShareButtons`)